### PR TITLE
Rename CODE_OWNERS -> Maintainers

### PR DESCRIPTION
**Pull request by AaronBallman**

### CODE REVIEW

**Comment by AaronBallman** [comment]
I've kept everyone in the same position they were previously in, just reworded for the newer nomenclature.

That said, I'd like explicit confirmation from @r4nt @njames93 and @sam-mccall that they're still willing and able to volunteer under the new maintainers guidance (https://llvm.org/docs/DeveloperPolicy.html#maintainers) before landing these changes. If you're not able to do so, that's totally fine! I can move you to the inactive maintainers part of the list and we can try to find other volunteers to step in to the role.

Also, I cannot seem to find a GitHub handle for Julie Hockett and her last community interactions seem to be from 2021. If someone can help find the correct information and tag her on the review, that would be appreciated.

**Comment by njames93** [comment]
I just no longer have the time needed to hold maintainer status. I'll still contribute as and when I have the time.

**Comment by AaronBallman** [comment]
> I just no longer have the time needed to hold maintainer status. I'll still contribute as and when I have the time.

Thank you for your help! I'll move you to the inactive maintainers list, and if you find you have more time in the future to step back into a maintenance role, we'll happily move you back to active status.

**Comment by AaronBallman** [comment]
Ping @r4nt and @sam-mccall

**Review by 5chmidti — APPROVED** [review]
LGTM in regard to my details.

There are some emails that have their `@` escaped. Was that intended?

**Review by HerrCai0907 — APPROVED** [review]

**Review by ilovepi — APPROVED** [review]
Thanks! My details seem correct.

**Review by HighCommander4 — APPROVED** [review]

**Review by kadircet — APPROVED** [review]

File: clang-tools-extra/Maintainers.txt
Comment (PeterChou1) [inline clang-tools-extra/Maintainers.txt:42]: my email changed since I graduated my new email is 
peterchou411@gmail.com

Comment (AaronBallman) [reply clang-tools-extra/Maintainers.txt]: Thanks! I've corrected this.

**Review by PeterChou1 — APPROVED** [review]

**Review by PiotrZSL — APPROVED** [review]

**Comment by 5chmidti** [comment]
@PiotrZSL The ping was in case you want to be added as a maintainer to clang-tidy as well. I’m mentioning it just to make sure.

**Comment by PiotrZSL** [comment]
> @PiotrZSL The ping was in case you want to be added as a maintainer to clang-tidy as well. I’m mentioning it just to make sure.

The designation of "maintainer" does not impact the established process. This role primarily introduces additional responsibilities without offering any demonstrable advantages. It is likely to become dead in the near future, mirroring the fate of the code owner role. At the end, being or not being a maintainer has no impact on me.

**Comment by HerrCai0907** [comment]
> At the end, being or not being a maintainer has no impact on me.
I think it will help the first time contributor in this sub project to find correct one who can  review their pr. But it is definitely not a  mandatory thing.

**Comment by AaronBallman** [comment]
> > @PiotrZSL The ping was in case you want to be added as a maintainer to clang-tidy as well. I’m mentioning it just to make sure.
> 
> The designation of "maintainer" does not impact the established process.

It does -- it helps newcomers figure out who can help them with reviews, answering questions, etc.

> This role primarily introduces additional responsibilities without offering any demonstrable advantages.

The only advantage to the maintainer is to get published credit for their increased responsibilities in the community, which isn't an advantage for everyone. The primary advantage is for the community.

> It is likely to become dead in the near future, mirroring the fate of the code owner role. 

The project has had this same concept in some form or fashion for at least 15 years now, so I don't think this is an accurate view.

> At the end, being or not being a maintainer has no impact on me.

There's no pressure to sign up to be one, but having a dedicated group of listed, active maintainers makes it more likely the project will thrive. FWIW, projects without active maintainers are starting to be removed or subsumed and I expect we'll continue to do that into the future.

File: clang-tools-extra/Maintainers.txt
Comment (PiotrZSL) [inline clang-tools-extra/Maintainers.txt:28]:
```suggestion

| Piotr Zegar
| me@piotrzegar.pl (email), PiotrZSL (GitHub), PiotrZSL (Discourse), PiotrZSL (Discord)

```

**Review by PiotrZSL — COMMENTED** [review]
Hmm, Then you may consider adding me.

**Review by llvm-beanz — APPROVED** [review]

**Pull request merged**
